pkg/scheduler/internal/cache/cache_test.go

```diff
@@ -1854,6 +1860,106 @@ func TestSchedulerCache_updateNodeInfoSnapshotList(t *testing.T) {
 	}
 }
 
+func TestSchedulerCache_ErrorCases(t *testing.T) {
+	cache := newCache(time.Second, time.Second, nil)
+	node := &v1.Node{ObjectMeta: metav1.ObjectMeta{Name: "test-node"}}
+	pod := &v1.Pod{ObjectMeta: metav1.ObjectMeta{Name: "test-pod"}}
+
+	testCases := []struct {
+		name          string
+		node          *v1.Node
+		pod           *v1.Pod
+		podOperation  func(*v1.Pod) error
+		nodeOperation func(*v1.Node) error
+	}{
+		{
+			name:         "assume unknown pod",
+			pod:          pod,
+			podOperation: cache.AssumePod,
+		},
+		{
+			name: "is unknown pod assumed",
+			pod:  pod,
+			podOperation: func(pod *v1.Pod) error {
+				_, err := cache.IsAssumedPod(pod)
+				return err
+			},
+		},
+		{
+			name:         "finish binding unknown pod",
+			pod:          pod,
+			podOperation: cache.FinishBinding,
+		},
+		{
+			name:         "forget unknown pod",
+			pod:          pod,
+			podOperation: cache.ForgetPod,
+		},
+		{
+			name: "get unknown pod",
+			pod:  pod,
+			podOperation: func(pod *v1.Pod) error {
+				_, err := cache.GetPod(pod)
+				return err
+			},
+		},
+		{
+			name: "update unknown pod",
+			pod:  pod,
+			podOperation: func(pod *v1.Pod) error {
+				return cache.updatePod(pod, pod)
+			},
+		},
+		{
+			name:         "remove unknown pod",
+			pod:          pod,
+			podOperation: cache.RemovePod,
+		},
+		{
+			name:          "remove unknown node",
+			node:          node,
+			nodeOperation: cache.RemoveNode,
+		},
+	}
+
+	for _, test := range testCases {
+		t.Run(test.name, func(t *testing.T) {
+			if test.nodeOperation != nil && test.node != nil {
+				if err := test.nodeOperation(test.node); err == nil {
+					t.Error("expected error, no error found")
+				}
+			}
+			if test.podOperation != nil && test.pod != nil {
+				if err := test.podOperation(test.pod); err == nil {
+					t.Error("expected error, no error found")
+				}
+			}
+		})
+	}
+
+	// should trigger an error log as there is no node info for this given name
```

Comment:
could this be verified? or else what's the point to leave these in the testcases?

Reply:
Resolved in 1c12a991112